ContentModification/EmbedFonts/EmbedFonts.cpp

```diff
@@ -1,194 +1,217 @@
 //
-// Copyright (c) 2010-2023, Datalogics, Inc. All rights reserved.
+// Copyright (c) 2010-2024, Datalogics, Inc. All rights reserved.
 //
 // This sample program demonstrates how to scan a PDF document to determine whether the fonts
-// used in that document are embedded. For each font the program finds that is not embedded in
-// the document, the sample looks for characters in that font that are used in the document.
-// It gathers up the characters and then requests that the Adobe PDF Library to generate a subset
-// font stream for these characters and then embed this font in a subset data stream. That is,
-// the program subsets the characters from that font into the PDF document.
+// used in that document are embedded.
 //
-// The default input PDF document has some fonts that are embedded and some that are not.
-// The sample program finds the fonts in the document that are not embedded, and then looks for
-// an alternative font on the local computer system to use as a substitute. Then, the program
-// subsets the characters of that font that are used in the document into that document, and
-// renames this newly subset font to show that it was embedded and subset.
 //
 
 #include "PSFCalls.h"
 #include "PERCalls.h"
 #include "PEWCalls.h"
-#include "PagePDECntCalls.h"
+#include "CosCalls.h"
 
 #include "InitializeLibrary.h"
 #include "APDFLDoc.h"
 
-#include "EmbedFonts.h"
-
 #define INPUT_DIR "../../../../Resources/Sample_Input/"
 #define INPUT_NAME "EmbedFonts-in.pdf"
 #define OUTPUT_NAME "EmbedFonts-out.pdf"
 
-int main(int argc, char **argv) {
-    APDFLib libInit; // Initialize the library
-    if (libInit.isValid() == false) {
-        ASErrorCode errCode = libInit.getInitError();
-        APDFLib::displayError(errCode);
-        return errCode;
-    }
-
-    std::string csInputFileName(argc > 1 ? argv[1] : INPUT_DIR INPUT_NAME);
-    std::string csOutputFileName(argc > 2 ? argv[2] : OUTPUT_NAME);
-    std::cout << "Will embed fonts into " << csInputFileName.c_str() << " and rewrite file as "
-              << csOutputFileName.c_str() << std::endl;
+#include <vector>
 
-    ContainerOfFonts vFontsInUse;
+// Tracking information for the fonts used in a document
+struct UnembeddedFont {
+    CosObj cosObj;
+};
 
+void EmbedSysFontForFontEntry(UnembeddedFont fontEntry, PDDoc pdDoc) {
     DURING
+        PDEFont pdeFontUnembedded = PDEFontCreateFromCosObj(&fontEntry.cosObj);
```

Comment:
There's indentation mismatch with the lines below it.

Reply:
fixed